lib/Reader/readerir.cpp

```diff
@@ -538,29 +526,26 @@ void GenIR::insertIRToKeepGenericContextAlive() {
     ContextLocalAddress = Arguments[MethodSignature.getTypeArgIndex()];
   }
 
-  // Indicate that the context location's address escapes by inserting a call
-  // to llvm.frameescape. Put that call just after the last alloca or the
-  // store to the scratch local.
-  LLVMBuilder->SetInsertPoint(InsertPoint->getNextNode());
-  Value *FrameEscape = Intrinsic::getDeclaration(JitContext->CurrentModule,
-                                                 Intrinsic::localescape);
-  Value *Args[] = {ContextLocalAddress};
-  const bool MayThrow = false;
-  makeCall(FrameEscape, MayThrow, Args);
-  // Don't move TempInsertionPoint up since what we added was not an alloca
-  LLVMBuilder->restoreIP(SavedInsertPoint);
-
   // This method now requires a frame pointer.
   // TargetMachine *TM = JitContext->TM;
   // TM->Options.NoFramePointerElim = true;
 
-  // TODO: we must convey the offset of this local to the runtime
-  // via the GC encoding.
-  // https://github.com/dotnet/llilc/issues/766
-
   if (JitContext->Options->DoInsertStatepoints) {
-    throw NotYetImplementedException("NYI: Generic Context reporting");
+    GcFuncInfo->GenericsContext = cast<AllocaInst>(ContextLocalAddress);
+  } else {
+    // Indicate that the context location's address escapes by inserting a call
+    // to llvm.frameescape. Put that call just after the last alloca or the
+    // store to the scratch local.
+    LLVMBuilder->SetInsertPoint(InsertPoint->getNextNode());
+    Value *FrameEscape = Intrinsic::getDeclaration(JitContext->CurrentModule,
+                                                   Intrinsic::localescape);
+    Value *Args[] = {ContextLocalAddress};
+    const bool MayThrow = false;
+    makeCall(FrameEscape, MayThrow, Args);
   }
+
+  // Don't move TempInsertionPoint up since what we added was not an alloca
+  LLVMBuilder->restoreIP(SavedInsertPoint);
```

Comment:
This is the kind of code that could be streamlined if there was always a GcFuncInfo around.